### PR TITLE
irdma: Add support for ibv_reg_dmabuf_mr

### PR DESCRIPTION
**Pull request by zhuyj**
Add support for ibv_reg_dmabuf_mr verb to the irdma provider.

Signed-off-by: Zhu Yanjun <yanjun.zhu@linux.dev>

### CODE REVIEW

**Comment by rleon** [comment]
You was asked to manage one PR and force push your changes.

Please don't close and open gazillion PRs.

Thanks

File: providers/irdma/uverbs.c [hunk 1]
```diff
@@ -151,6 +151,29 @@ struct ibv_mr *irdma_ureg_mr(struct ibv_pd *pd, void *addr, size_t length,
...
+				    int access)
+{
+	struct irdma_umr *umr;
+	int err;
+
+	umr = malloc(sizeof(*umr));
+	if (!umr)
+		return NULL;
+
+	memset(umr, 0, sizeof(*umr));
```
Comment (shirazsaleem) [inline providers/irdma/uverbs.c:165]: Just calloc umr obj instead?

**Comment by zhuyj** [comment]
> 

Got it.

**Comment by zhuyj** [comment]
> You were asked to manage one PR and force push your changes.
> 
> Please don't close and open gazillion PRs.
> 
> Thanks

Got it. This is the latest pull request.

**Pull request merged**